# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/new-regexp/RegExpAPI.cpp
# Commit: 18b93628199b
# Full Hash: 18b93628199b312a826b426162d0a43883b2edae
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-06 21:51:14
# Regressor Bug: 1634135
# File Overlap Count: 1
# Description:
#   Bug 1634135: Disable named capture parsing until fully supported r=mgaudet
#   
#   The engine supports parsing named captures, but we don't have the code in place to expose the captured groups. Until we do, this code will make sure that we get a syntax error when parsing them.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73115
# ==============================================================================

diff -r 6aaf6e87a5d1 -r 18b93628199b js/src/new-regexp/RegExpAPI.cpp
--- a/js/src/new-regexp/RegExpAPI.cpp	Tue May 05 12:49:53 2020 +0000
+++ b/js/src/new-regexp/RegExpAPI.cpp	Tue May 05 12:49:54 2020 +0000
@@ -243,6 +243,14 @@
     ReportSyntaxError(ts, result, chars.begin().get(), chars.length());
     return false;
   }
+  if (!result.capture_name_map.is_null()) {
+    // We can parse named captures, but we don't support them yet.
+    // Until we do, this will report a syntax error:
+    //  "invalid capture group name in regular expression"
+    result.error = RegExpError::kInvalidCaptureGroupName;
+    ReportSyntaxError(ts, result, chars.begin().get(), chars.length());
+    return false;
+  }
   return true;
 }
 
@@ -254,6 +262,14 @@
     ReportSyntaxError(ts, result, pattern);
     return false;
   }
+  if (!result.capture_name_map.is_null()) {
+    // We can parse named captures, but we don't support them yet.
+    // Until we do, this will report a syntax error:
+    //  "invalid capture group name in regular expression"
+    result.error = RegExpError::kInvalidCaptureGroupName;
+    ReportSyntaxError(ts, result, pattern);
+    return false;
+  }
   return true;
 }
 